File: HelpfulSense/app/src/main/java/com/andruid/magic/helpfulsense/adapter/ViewPagerAdapter.java
package com.andruid.magic.helpfulsense.adapter;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import com.andruid.magic.helpfulsense.fragment.AlertFragment;
import com.andruid.magic.helpfulsense.fragment.ContactsFragment;
import com.andruid.magic.helpfulsense.fragment.MessageFragment;
import com.andruid.magic.helpfulsense.fragment.SettingsFragment;

import static com.andruid.magic.helpfulsense.data.Constants.NO_OF_TABS;

public class ViewPagerAdapter extends FragmentPagerAdapter {
    private static final int POS_ALERT = 0, POS_MESSAGE = 1, POS_CONTACTS = 2;

    public ViewPagerAdapter(FragmentManager fm) {
        super(fm, FragmentPagerAdapter.BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT);
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        Fragment fragment;
        switch (position){
            case POS_ALERT:
                fragment = AlertFragment.newInstance();
                break;
            case POS_MESSAGE:
                fragment = MessageFragment.newInstance();
                break;
            case POS_CONTACTS:
                fragment = ContactsFragment.newInstance();
                break;
            default:
                fragment = SettingsFragment.newInstance();
                break;
        }
        return fragment;
    }

    @Override
    public int getCount() {
        return NO_OF_TABS;
    }
}

File: HelpfulSense/app/src/main/java/com/andruid/magic/helpfulsense/ui/adapter/ViewPagerAdapter.kt
package com.andruid.magic.helpfulsense.ui.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.andruid.magic.helpfulsense.data.NO_OF_TABS
import com.andruid.magic.helpfulsense.ui.fragment.AlertFragment
import com.andruid.magic.helpfulsense.ui.fragment.ContactsFragment
import com.andruid.magic.helpfulsense.ui.fragment.MessageFragment
import com.andruid.magic.helpfulsense.ui.fragment.SettingsFragment

class ViewPagerAdapter(fragmentActivity: FragmentActivity) : FragmentStateAdapter(fragmentActivity) {
    companion object {
        private const val POS_ALERT = 0
        private const val POS_MESSAGE = 1
        private const val POS_CONTACTS = 2
    }

    override fun getItemCount() = NO_OF_TABS

    override fun createFragment(position: Int): Fragment {
        return when (position) {
            POS_ALERT -> AlertFragment.newInstance()
            POS_MESSAGE -> MessageFragment.newInstance()
            POS_CONTACTS -> ContactsFragment.newInstance()
            else -> SettingsFragment.newInstance()
        }
    }
}